### PR TITLE
[bashible] drop kernel management

### PR DESCRIPTION
**Pull request by RomanenkoDenys**
Signed-off-by: Denis Romanenko <denis.romanenko@flant.com>

## Description

Now we cannot have ideas about how to proper management kernel versions on the nodes.

## Why do we need it, and what problem does it solve?

Kernel management completely removed.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [x] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: candi
type: feature
summary: Kernel version management removed.
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/azure'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/vsphere'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/azure'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by z9r5** [comment]
Changed impact_level to default.

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/azure'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/openstack'...

File: modules/040-node-manager/docs/FAQ_RU.md [hunk 1]
```diff
@@ -491,6 +491,17 @@ spec:
 
 Дополнительные шаги для конфигурации узлов задаются при помощи custom resource `NodeGroupConfiguration`.
 
+## Как обновить ядро линукс на узлах до предпочитаемой версии ?
+
+Установить для ноды аннотацию `update.node.deckhouse.io/force-install-desired-kernel`.
```
Comment (z9r5) [inline modules/040-node-manager/docs/FAQ_RU.md:496]:
```suggestion
## Как обновить ядро Linux на узлах до рекомендуемой версии?

Версия ядра на узле может отличаться от рекомендуемой, но попадать в список допустимых версий. В этом случае обновления ядра на узле не произойдет. Чтобы принудительно обновить ядро до рекомендуемой версии, нужно установить на узле аннотацию `update.node.deckhouse.io/force-install-desired-kernel`.
```

File: modules/040-node-manager/docs/FAQ_RU.md [hunk 2]
```diff
@@ -491,6 +491,17 @@ spec:
...
+
+```shell
+kubectl annotate node <имя узла> update.node.deckhouse.io/force-install-desired-kernel=
+
```
Comment (z9r5) [inline modules/040-node-manager/docs/FAQ_RU.md:500]:
```suggestion
kubectl annotate node <имя узла> update.node.deckhouse.io/force-install-desired-kernel=
```

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/gcp'...

File: modules/040-node-manager/docs/FAQ.md [hunk 3]
```diff
@@ -244,17 +244,116 @@ Create a `NodeGroup` with the following parameters:
 ```yaml
 cri:
   type: NotManaged
-operatingSystem:
-  manageKernel: false
 ```
 
 Then put the node under the control of `node-manager`.
 
+## How do I update kernel on nodes ?
```
Comment (z9r5) [inline modules/040-node-manager/docs/FAQ.md:251]:
```suggestion
## How do I update kernel on nodes?
```

File: modules/040-node-manager/docs/FAQ_RU.md [hunk 4]
```diff
@@ -244,17 +244,116 @@ journalctl -fu bashible
 ```yaml
 cri:
   type: NotManaged
-operatingSystem:
-  manageKernel: false
 ```
 
 После чего добавить узел под управление `node-manager`.
 
+## Как обновить ядро на узлах ?
```
Comment (z9r5) [inline modules/040-node-manager/docs/FAQ_RU.md:251]:
```suggestion
## Как обновить ядро на узлах?
```

File: modules/040-node-manager/docs/FAQ_RU.md [hunk 5]
```diff
@@ -244,17 +244,116 @@ journalctl -fu bashible
...
+
+### Для дистрибутивов, основанных на Debian
+
+Примените ресурс `NodeGroupConfiguration`, поле `desired_version` должно содержать желаемую версию ядра:
```
Comment (z9r5) [inline modules/040-node-manager/docs/FAQ_RU.md:255]:
```suggestion
Создайте ресурс `NodeGroupConfiguration`, указав в поле `desired_version` желаемую версию ядра:
```

Comment (z9r5) [reply modules/040-node-manager/docs/FAQ_RU.md:255]: Это получается не параметр ресурса а прям в bash-скрипте в content надо править? А давай копирайты уберем из скрипта?  @RomanenkoDenys

File: modules/040-node-manager/docs/FAQ_RU.md [hunk 6]
```diff
@@ -244,17 +244,116 @@ journalctl -fu bashible
...
+  
+    if [[ "$version_in_use" == "$desired_version" ]]; then
+      exit 0
+    fi
+  
+    bb-deckhouse-get-disruptive-update-approval
+    bb-apt-install "linux-image-${desired_version}"
+```
+
+### Для дистрибутивов, основанных на Centos
```
Comment (z9r5) [inline modules/040-node-manager/docs/FAQ_RU.md:303]:
```suggestion
### Для дистрибутивов, основанных на CentOS
```

File: modules/040-node-manager/docs/FAQ_RU.md [hunk 7]
```diff
@@ -244,17 +244,116 @@ journalctl -fu bashible
...
+
+Примените ресурс `NodeGroupConfiguration`, поле `desired_version` должно содержать желаемую версию ядра:
```
Comment (z9r5) [inline modules/040-node-manager/docs/FAQ_RU.md:305]:
```suggestion
Создайте ресурс `NodeGroupConfiguration`, указав в поле `desired_version` желаемую версию ядра:
```

File: modules/040-node-manager/docs/FAQ.md [hunk 8]
```diff
@@ -244,17 +244,116 @@ Create a `NodeGroup` with the following parameters:
...
-  manageKernel: false
 ```
 
 Then put the node under the control of `node-manager`.
 
+## How do I update kernel on nodes?
+
+### Debian-based distros
+
+Deploy `NodeGroupConfiguration` scripts, fill `desired_version` with desired kernel version:
```
Comment (z9r5) [inline modules/040-node-manager/docs/FAQ.md:255]:
```suggestion
Deploy `NodeGroupConfiguration` scripts, fill `desired_version` with desired kernel version:
Create a `Node Group Configuration` resource by specifying the desired kernel version in the `desired_version' variable of the shell script (the resource's content parameter):
```

File: modules/040-node-manager/docs/FAQ.md [hunk 9]
```diff
@@ -244,17 +244,116 @@ Create a `NodeGroup` with the following parameters:
...
+  
+    if [[ "$version_in_use" == "$desired_version" ]]; then
+      exit 0
+    fi
+  
+    bb-deckhouse-get-disruptive-update-approval
+    bb-apt-install "linux-image-${desired_version}"
+```
+
+### Centos-based distros
```
Comment (z9r5) [inline modules/040-node-manager/docs/FAQ.md:303]:
```suggestion
### CentOS-based distros
```

File: modules/040-node-manager/docs/FAQ.md [hunk 10]
```diff
@@ -244,17 +244,116 @@ Create a `NodeGroup` with the following parameters:
...
+
+Deploy `NodeGroupConfiguration` scripts, fill `desired_version` with desired kernel version:
```
Comment (z9r5) [inline modules/040-node-manager/docs/FAQ.md:305]:
```suggestion
Create a `Node Group Configuration` resource by specifying the desired kernel version in the `desired_version` variable of the shell script (the resource's content parameter):
```

**Review by z9r5 — DISMISSED** [review]

File: modules/040-node-manager/docs/FAQ_RU.md (hunk 7 above)
Comment (apolovov) [reply modules/040-node-manager/docs/FAQ_RU.md:305]: параметр content ресурса —> spec.content

File: modules/040-node-manager/docs/FAQ_RU.md [hunk 11]
```diff
@@ -244,17 +244,116 @@ journalctl -fu bashible
...
+
+```yaml
+apiVersion: deckhouse.io/v1alpha1
+kind: NodeGroupConfiguration
+metadata:
+  name: install-kernel.sh
+  resourceVersion: "7116"
+  uid: a23bf2fc-1466-4556-a00a-346a64d2ba9f
```
Comment (apolovov) [inline modules/040-node-manager/docs/FAQ_RU.md:313]:
```suggestion
```

File: modules/040-node-manager/docs/FAQ_RU.md (hunk 5 above)
Comment (apolovov) [reply modules/040-node-manager/docs/FAQ_RU.md:255]: параметр content ресурса —> spec.content

File: modules/040-node-manager/docs/FAQ.md (hunk 8 above)
Comment (apolovov) [reply modules/040-node-manager/docs/FAQ.md:255]: the resource's content parameter -> spec.content

File: modules/040-node-manager/docs/FAQ.md (hunk 10 above)
Comment (apolovov) [reply modules/040-node-manager/docs/FAQ.md:305]: the resource's content parameter -> spec.content

**Review by z9r5 — APPROVED** [review]

**Review by apolovov — APPROVED** [review]

**Review by yalosev — APPROVED** [review]

**Pull request merged**